Popolo url

octopusinvitro opened the pull request:
This PR uses the `popolo_url` field from the data, instead of building the popolo url manually.

It updates the relevant urls in the tests. The `test_retrieving_popolo` test had to be updated with `:record => :new_episodes` so that the URL to fetch from is updated.

Closes #11

tmtmtmtm commented:
If I'm understanding this correctly, VCR is failing because it's using the default 'only ever record once' mode, but here it's being asked to fetch from a new URL. If so, then I think we should simply change the config option to ['new episodes'](https://relishapp.com/vcr/vcr/v/1-7-0/docs/record-modes/new-episodes) instead, rather than blowing away the existing cassette.

tmtmtmtm commented:
As this is also a user visible change (albeit one that _should_ be backwards compatible in the vast majority of cases) we need to make sure it's signposted in the CHANGELOG (and should trigger a new version)

octopusinvitro commented:
@tmtmtmtm I think this is ready to autosquash and merge if you don't spot anything:
- I made a fixup commit that hopefully will recover the previous `popolo.yml` file and also adds the `:record => :new_episodes` to the relevant test
- Updated changelog and version from 0.9.0 to 1.0.0 :scream: :scream_cat: 
- Updated the PR description to reflect the fixup-commit changes

tmtmtmtm commented:
@octopusinvitro I don't think the entries in the CHANGELOG are quite right here — this file is primarily for consumers of the gem to read to find out what things have changed that will affect them. They generally won't care about inward-facing implementation details (if they do, they can pay attention to the commit log instead). The only relevant thing here to them is that `popolo_url` is now the version from `everypolitican-data` directly, and thus from a different host than before.

Also, why did you go to 1.0.0 for this? I'm not sure this is ready for that yet. This seems much more like a 0.10.0 to me…

octopusinvitro commented:
@tmtmtmtm Good point on the changelog, and lol, for some reason my brain thought that 1.0 is what comes after 0.9 :laughing: :laughing: :laughing: 
Fixed :+1:

tmtmtmtm commented on test/everypolitician_test.rb:
```diff
@@ -100,7 +100,7 @@ def test_alternative_constant_name
 
   def test_retrieving_popolo
-    VCR.use_cassette('popolo') do
+    VCR.use_cassette('popolo', :record => :new_episodes) do
```
I was a little surprised that rubocop didn't complain about you using the old-fashioned hash syntax here (rather than `record: :new_episodes`), but that'll be because we don't actually _have_ rubocop running here yet! 

I'll add that today, which will auto-fix this, so no need to change it here, but worth noting for future reference…

octopusinvitro replied:
Great catch! I usually write these things as `record: :new_episodes`, so shame on me for copy-pasting from the webpage. Great idea to introduce rubocop! :+1:

tmtmtmtm commented:
Looks good to me! 👍
